include/Note.h

```diff
@@ -244,7 +244,14 @@ class LMMS_EXPORT Note : public SerializingObject
 };
 
 
-typedef QVector<Note *> NoteVector;
+
+
+class NoteVector: public QVector<Note*>
+{
+public:
+	bool getBounds(TimePos& start, TimePos& end, int& lower, int& upper) const;
+	bool transpose(int semitones) const;
```

Comment:
So I should make them static methods of `NoteVector` instead?

Reply:
I would leave `NoteVector` as a typedef for `QVector<Note*>`, and make them free (non-member) functions, so
```cpp
// See also my later comment for further changes to getBounds
bool getBounds(const NoteVector& notes, TimePos& start, TimePos& end, int& lower, int& upper);
bool transpose(const NoteVector& notes, int semitones);
```

Reply:
I don't see a lot of these kind of functions in header files (I want at least `getBounds` to be accessible), so I figured putting it in a class would make things cleaner. Or is there some other way to not have a ton of functions in the global scope?

Reply:
> Or is there some other way to not have a ton of functions in the global scope?

#6174?

> I don't see a lot of these kind of functions in header files (I want at least getBounds to be accessible), so I figured putting it in a class would make things cleaner.

I really wouldn't worry about it too much. You say you want to make the functions reusable - perhaps the reason there aren't many free functions in header files is because we haven't been very good at making things reusable up until now. If you look in namespace `std`, there are loads of free functions, but that doesn't cause anyone a problem. If you want, you can add `notes` or something to the names to make the purpose clearer without a class name for context (e.g. `boundsForNotes`, `transposeNotes`), but even without, you won't have problems with name collisions since parameter types distinguish functions in C++ (unlike in C, Python, etc.).

Reply:
Ok, that makes sense. I moved out `getBounds` and renamed it `boundsForNotes`. And `transpose` is so simple that I just dropped it.